src/test/java/com/microsoft/sqlserver/jdbc/datatypes/SQLVariantResultSetTest.java

```diff
@@ -1061,6 +1061,16 @@ public void testTimeClassAsSqlVariant() throws SQLException {
             assertEquals(object.getClass(), java.sql.Time.class);;
         }
     }
+    
+    @Test
+    public void testCastThenGetNumeric() throws SQLException {
```

Comment:
Please add test cases for rest of the datatypes this PR applies to.

Reply:
Done (applies to other comments too)